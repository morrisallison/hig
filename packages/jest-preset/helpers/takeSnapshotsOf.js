const React = require("react");
const renderer = require("react-test-renderer");

/**
 * @typedef {import("react-test-renderer").TestRendererOptions} Options
 */

/**
 * @typedef {Object} Case
 * @property {string} desc
 * @property {Object} [props]
 * @property {import("react").ReactElement} [subject]
 */

/** @type {Options} */
const defaultOptions = {
  createNodeMock(element) {
    return document.createElement(element.type);
  }
};

/**
 * @param {import("react").Component|null} Component
 * @param {Case[]} cases
 * @param {Options} [options]
 */
module.exports = function takeSnapshotsOf(Component, cases, options) {
  cases.forEach(({ desc, props, subject }) => {
    it(desc, () => {
      const rendererOptions = Object.assign({}, defaultOptions, options);
      const vDom = subject || React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
