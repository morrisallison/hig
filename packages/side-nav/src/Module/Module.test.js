import React from "react";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import { mount } from "enzyme";

import Module from "./Module";

describe("side-nav/Module", () => {
  describe("onClickTitle", () => {
    it("adds click handler to title", () => {
      const onClickTitle = jest.fn();
      const wrapper = mount(
        <Module title="Module" onClickTitle={onClickTitle} />
      );

      wrapper.find(".hig__side-nav__module__link__title").simulate("click");
      expect(onClickTitle).toHaveBeenCalled();
    });
  });

  describe("onClickCollapseButton", () => {
    it("adds click handler to collapse button", () => {
      const onClickCollapseButton = jest.fn();
      const wrapper = mount(
        <Module title="Module" onClickCollapseButton={onClickCollapseButton}>
          Content
        </Module>
      );

      wrapper.find(".hig__side-nav__module__collapse-button").simulate("click");
      expect(onClickCollapseButton).toHaveBeenCalled();
    });
  });

  describe("snapshot tests", () => {
    takeSnapshotsOf(Module, [
      {
        desc: "renders with minimal props",
        props: {
          title: "Module"
        }
      },
      {
        desc: "renders an external link",
        props: {
          link: "http://example.com",
          target: "_blank",
          title: "Module"
        }
      },
      {
        desc: "renders a link",
        props: {
          link: "http://example.com",
          title: "Module"
        }
      },
      {
        desc: "renders a expanded menu with active children",
        props: {
          activeChildren: true,
          children: <div data-test="children" />,
          icon: <div data-test="icon" />,
          title: "Module"
        }
      },
      {
        desc: "renders with a click handlers",
        props: {
          onClickCollapseButton: function handleClickCollapseButton() {},
          onClickTitle: function handleClickTitle() {},
          title: "Module"
        }
      },
      {
        desc: "renders an active button",
        props: {
          active: true,
          icon: <span data-test="icon" />,
          onClickCollapseButton: function handleClickCollapseButton() {},
          onClickTitle: function handleClickTitle() {},
          title: "Module"
        }
      }
    ]);
  });
});
