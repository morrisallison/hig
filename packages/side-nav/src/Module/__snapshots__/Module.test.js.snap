// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`side-nav/Module snapshot tests renders a expanded menu with active children 1`] = `
<div
  className="hig--light-theme hig__side-nav__module"
  onMouseOver={undefined}
>
  <div
    className="hig__side-nav__module__row"
  >
    <div
      className="hig--light-theme hig__side-nav__module__link hig__side-nav__module__link--active-children"
      href={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="-1"
      target={undefined}
    >
      <div
        className="hig__side-nav__module__link__icon"
      >
        <div
          data-test="icon"
        />
      </div>
      <div
        className="hig__side-nav__module__link__title"
      >
        Module
      </div>
    </div>
    <div
      aria-pressed={true}
      className="hig--light-theme hig__side-nav__module__collapse-button"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="hig__icon hig__icon--24-size"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg width=\\"10\\" height=\\"6\\" viewBox=\\"0 0 10 6\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"caret-24\\"><polyline transform=\\"rotate(45 9.828 3)\\" points=\\"8 2 8 8 2 8\\" stroke=\\"#2E4258\\" stroke-width=\\"1.15\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"/></svg>",
          }
        }
      />
    </div>
  </div>
  <div
    className="hig__side-nav__module__submodules"
  >
    <div
      data-test="children"
    />
  </div>
</div>
`;

exports[`side-nav/Module snapshot tests renders a link 1`] = `
<div
  className="hig--light-theme hig__side-nav__module"
  onMouseOver={undefined}
>
  <div
    className="hig__side-nav__module__row"
  >
    <a
      className="hig--light-theme hig__side-nav__module__link"
      href="http://example.com"
      onClick={[Function]}
      onKeyDown={[Function]}
      role={undefined}
      tabIndex="0"
      target="_self"
    >
      <div
        className="hig__side-nav__module__link__icon"
      />
      <div
        className="hig__side-nav__module__link__title"
      >
        Module
      </div>
    </a>
  </div>
  <div
    className="hig__side-nav__module__submodules hig__side-nav__module__submodules--no-icon"
  />
</div>
`;

exports[`side-nav/Module snapshot tests renders an active button 1`] = `
<div
  className="hig--light-theme hig__side-nav__module"
  onMouseOver={undefined}
>
  <div
    className="hig__side-nav__module__row"
  >
    <div
      className="hig--light-theme hig__side-nav__module__link hig__side-nav__module__link--active"
      href={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      target={undefined}
    >
      <div
        className="hig__side-nav__module__link__icon"
      >
        <span
          data-test="icon"
        />
      </div>
      <div
        className="hig__side-nav__module__link__title"
      >
        Module
      </div>
    </div>
  </div>
  <div
    className="hig__side-nav__module__submodules"
  />
</div>
`;

exports[`side-nav/Module snapshot tests renders an external link 1`] = `
<div
  className="hig--light-theme hig__side-nav__module"
  onMouseOver={undefined}
>
  <div
    className="hig__side-nav__module__row"
  >
    <a
      className="hig--light-theme hig__side-nav__module__link"
      href="http://example.com"
      onClick={[Function]}
      onKeyDown={[Function]}
      role={undefined}
      tabIndex="0"
      target="_blank"
    >
      <div
        className="hig__side-nav__module__link__icon"
      />
      <div
        className="hig__side-nav__module__link__title"
      >
        Module
      </div>
      <div
        className="hig__side-nav__module__link__external-link-icon"
      >
        <div
          className="hig__icon hig__icon--24-size"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg width=\\"9\\" height=\\"8\\" viewBox=\\"0 0 9 8\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"external-link-24\\"><g stroke=\\"#2E4258\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><polyline points=\\"7.7074 6.4303 7.7074 0.5003 1.7904 0.5003\\"/><path d=\\"M1.08617072,7.11062928 L7.6864,0.5104\\"/></g></svg>",
            }
          }
        />
      </div>
    </a>
  </div>
  <div
    className="hig__side-nav__module__submodules hig__side-nav__module__submodules--no-icon"
  />
</div>
`;

exports[`side-nav/Module snapshot tests renders with a click handlers 1`] = `
<div
  className="hig--light-theme hig__side-nav__module"
  onMouseOver={undefined}
>
  <div
    className="hig__side-nav__module__row"
  >
    <div
      className="hig--light-theme hig__side-nav__module__link"
      href={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      target={undefined}
    >
      <div
        className="hig__side-nav__module__link__icon"
      />
      <div
        className="hig__side-nav__module__link__title"
      >
        Module
      </div>
    </div>
  </div>
  <div
    className="hig__side-nav__module__submodules hig__side-nav__module__submodules--no-icon"
  />
</div>
`;

exports[`side-nav/Module snapshot tests renders with minimal props 1`] = `
<div
  className="hig--light-theme hig__side-nav__module"
  onMouseOver={undefined}
>
  <div
    className="hig__side-nav__module__row"
  >
    <div
      className="hig--light-theme hig__side-nav__module__link"
      href={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      target={undefined}
    >
      <div
        className="hig__side-nav__module__link__icon"
      />
      <div
        className="hig__side-nav__module__link__title"
      >
        Module
      </div>
    </div>
  </div>
  <div
    className="hig__side-nav__module__submodules hig__side-nav__module__submodules--no-icon"
  />
</div>
`;
